### PR TITLE
Stabilize `defaultBlock`, `directInsert` API's and `getDirectInsertBlock` selector

### PR DESCRIPTION
**Pull request by sethrubenstein**
…elector

## What?
This PR stabilizes the `__experimentalDefaultBlock`, `__experimentalDirectInsert` API's and `__experimentalGetDirectInsertBlock` selector.

## Why?
This stabilizes `defaultBlock` so that it can be used by #50982 and https://github.com/WordPress/gutenberg/pull/51026. Solves for https://github.com/WordPress/gutenberg/issues/52068

## How?
Updated references to API's and added deprecation notices to all affected API's and selectors.

## Testing Instructions
1. Everything should work as is.

### CODE REVIEW

**Review by scruffian — APPROVED** [review]
This looks good and works as expected. I'll wait for someone else to confirm that making these APIs public is a good idea before merging.

**Comment by getdave** [comment]
🚀

**Comment by sethrubenstein** [comment]
Woot, thanks all! I'll work on getting modified https://github.com/WordPress/gutenberg/issues/50982 to reflect these new changes.

File: packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js [hunk 1]
```diff
@@ -108,11 +115,29 @@ export default function useNestedSettingsUpdate(
 		}
 
 		if ( __experimentalDefaultBlock !== undefined ) {
-			newSettings.__experimentalDefaultBlock = __experimentalDefaultBlock;
+			deprecated( '__experimentalDefaultBlock', {
+				alternative: 'defaultBlock',
+				since: '6.3',
+				version: '6.4',
```
Comment (youknowriad) [inline packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js:121]: Hey, should we follow up on these deprecations and act on them (remove or extend) as we've reached the version.

**Pull request merged**